src/auth.rs

```diff
@@ -87,66 +88,120 @@ pub fn get_hash<T: Digest>(text: &str) -> Vec<u8> {
     hasher.finalize().to_vec()
 }
 
-impl Middleware<crate::MiniserveConfig> for Auth {
-    fn response(
-        &self,
-        req: &HttpRequest<crate::MiniserveConfig>,
-        resp: HttpResponse,
-    ) -> Result<Response> {
-        let required_auth = &req.state().auth;
+pub struct AuthMiddleware<S> {
```

Comment:
While I recognize that this is a direct port of the pre-existing middleware from actix-web 0.x, perhaps by now it might be a better idea to use the official [actix-web-httpauth](https://github.com/actix/actix-extras/tree/master/actix-web-httpauth)? It seems to be flexible enough as you can pass a custom authenticator to it. It might save us from some complicated-looking code. :)

Can you look at it and see whether it's a good fit?

Reply:
Thanks for the suggestion, I'll take a look at it.

Reply:
Done in 8bf16f6.

Reply:
Just looked at it. Super great change! Really cuts down on some very complicated looking code. Let's face it, actix-web middlewares just aren't pretty. :D